src/persist/src/indexed/mod.rs

```diff
@@ -941,6 +939,22 @@ impl AppliedState {
         Ok(())
     }
 
+    /// Physically deletes any blobs queued for deletion.
+    ///
+    /// NB: This is called within `apply_unbatched_cmd`, which assumes
+    /// idempotent usage of external systems. The [Blob] trait returns success
```

Comment:
nit: this comment seems a little bit incomplete, in that this only feels idempotent if keys are truly write once, (otherwise you could write (foo, a), delete foo, write (foo, b) delete foo - which doesn't line up with my intuition re: idempotence?)

I think its fine to just add a bit about how we use uuids for keys to make any accidental key collisions extremely unlikely, and so keys can be assumed to be write-once